Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.py

```diff
@@ -0,0 +1,651 @@
+                    demisto.debug(f"Going to async sleep for {to_sleep} seconds to avoid rate limit error")
+                    await asyncio.sleep(int(to_sleep))
+                else:
+                    # if the RESET value does not exist in the header then
+                    # sleep for default 1 second as the rate limit remaining is 0
+                    demisto.debug("Did not find a rate limit reset value, going to sleep for 1 second to avoid rate limit error")
+                    await asyncio.sleep(1)
+
+                return True
+
+    except ValueError as ve:
+        logging.error(f"Value error when honoring the rate limiting wait time {headers} {str(ve)}")
+
+    return False
+
+
+async def handle_event_type_async(
+    client: Client,
+    event_type: str,
+    start_time: str,
+    end_time: str,
+    offset: int,
+    limit: int,
+    send_to_xsiam: bool,
+    is_re_fetch_failed_fetch: bool = False,
+) -> tuple[str, dict]:
+    page_size = min(limit, MAX_EVENTS_PAGE_SIZE)
+    params = assign_params(limit=page_size, offset=offset, **get_time_window_params(event_type, start_time, end_time))
+
+    demisto.debug(f"Fetching '{event_type}' events with params: {params}")
+    # If this is a retry of a previous failure, log it.
+    if is_re_fetch_failed_fetch:
+        demisto.debug(f"Retrying failed fetch for type={event_type}, params={params}")
+    success_res, failures = await fetch_and_send_events_async(
+        client, event_type, params, limit, send_to_xsiam, is_re_fetch_failed_fetch
+    )
+    if is_re_fetch_failed_fetch and success_res:
+        demisto.debug(f"Retry succeeded for type={event_type}, params={params}")
+
+    if not success_res and failures and not is_re_fetch_failed_fetch:
+        # if there are no success fetch/send, raise an exception and keep the previous next_fetch_start_time
+        e: DemistoException = failures[0]
+        demisto.error(f"Failed to fetch events for type={event_type}, params={params}: {str(e)}")
+        if hasattr(e, "exception") and hasattr(e.exception, "status"):
+            demisto.error(f"HTTP status: {getattr(e.exception, 'status', None)}")
+        if hasattr(e, "res"):
+            demisto.error(f"API response: {getattr(e, 'res', None)}")
+        if "Unauthorized" in e.message:
+            msg = "Unauthorized Error: please validate your credentials."
+        elif "certificate verify failed" in e.message:
+            msg = "Connection Error: certificate verification failed, try to use the insecure checkbox."
+        elif "Cannot connect to host" in e.message:
+            msg = "Connection Error: please validate your Server URL."
+        else:
+            msg = (
+                f"Fetching event_type_start_time_end_time_offset: {event_type}_{start_time}_{end_time}_{offset} "
+                f"Failed, {str(failures[0])}"
+            )
+        demisto.error(msg)
+        raise DemistoException(msg, exception=failures[0])
```

Comment:
Consider handling the error according to status code and not the message received.

Reply:
Imadded a combination, code status or message